panther_gpiod/test/test_gpiod.cpp

```diff
@@ -0,0 +1,208 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <streambuf>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include "gpiod.hpp"
+
+#include "panther_gpiod/gpio_driver.hpp"
+
+using namespace panther_gpiod;
+
+class TestGPIODriver : public ::testing::Test
+{
+public:
+  TestGPIODriver() = default;
+  virtual ~TestGPIODriver() = default;
+
+  void GPIOEventCallback(const GPIOInfo & gpio_info);
+
+protected:
+  void ReinitializeGPIODriver(std::vector<GPIOInfo> gpio_info_storage);
+  void ReinitializeEventSummary();
+  void SetAndVerifyPinState(const GPIOPin & pin);
+
+  std::unique_ptr<GPIODriver> gpio_driver_;
+  std::pair<GPIOPin, gpiod::line::value> last_gpio_event_summary_;
+  const std::vector<GPIOInfo> gpio_config_info_{
+    GPIOInfo{GPIOPin::LED_SBC_SEL, gpiod::line::direction::OUTPUT},
+  };
+};
+
+void TestGPIODriver::ReinitializeGPIODriver(std::vector<GPIOInfo> gpio_info_storage)
+{
+  gpio_driver_.reset();
+  gpio_driver_ = std::make_unique<GPIODriver>(gpio_info_storage);
+}
```

Comment:
Yes, it's a better solution, but I suggest using `SetUp`. Google Test does not capture exceptions thrown in constructors. If an exception is thrown in `SetUp`, it will be caught, and the test will be marked as failed.